pkg/querier/queryrange/codec_test.go

```diff
@@ -284,9 +289,456 @@ func Test_codec_DecodeResponse(t *testing.T) {
+						]
+					}
+				],
+				"stats" : {
+					"ingester" : {
+						"store": {
+							"chunksDownloadTime": 0,
+							"totalChunksRef": 0,
+							"totalChunksDownloaded": 0,
+							"chunk" :{
+								"compressedBytes": 0,
+								"decompressedBytes": 0,
+								"decompressedLines": 0,
+								"headChunkBytes": 0,
+								"headChunkLines": 0,
+								"totalDuplicates": 0
+							}
+						},
+						"totalBatches": 0,
+						"totalChunksMatched": 0,
+						"totalLinesSent": 0,
+						"totalReached": 0
+					},
+					"querier": {
+						"store": {
+							"chunksDownloadTime": 0,
+							"totalChunksRef": 0,
+							"totalChunksDownloaded": 0,
+							"chunk" :{
+								"compressedBytes": 0,
+								"decompressedBytes": 0,
+								"decompressedLines": 0,
+								"headChunkBytes": 0,
+								"headChunkLines": 0,
+								"totalDuplicates": 0
+							}
+						}
+					},
+					"cache": {
+						"chunk": {
+							"entriesFound": 0,
+							"entriesRequested": 0,
+							"entriesStored": 0,
+							"bytesReceived": 0,
+							"bytesSent": 0,
+							"requests": 0,
+							"downloadTime": 0
+						},
+						"index": {
+							"entriesFound": 0,
+							"entriesRequested": 0,
+							"entriesStored": 0,
+							"bytesReceived": 0,
+							"bytesSent": 0,
+							"requests": 0,
+							"downloadTime": 0
+						},
+						"result": {
+							"entriesFound": 0,
+							"entriesRequested": 0,
+							"entriesStored": 0,
+							"bytesReceived": 0,
+							"bytesSent": 0,
+							"requests": 0,
+							"downloadTime": 0
+						}
+					},
+					"summary": {
+						"bytesProcessedPerSecond": 0,
+						"execTime": 0,
+						"linesProcessedPerSecond": 0,
+						"queueTime": 0,
+                        "shards": 0,
+                        "splits": 0,
+						"subqueries": 0,
+						"totalBytesProcessed":0,
+                                                "totalEntriesReturned":0,
+						"totalLinesProcessed":0
+					}
+				}
+			}
+		}`,
+		},
+		// vector test
+		{
+			promql.Vector{
+				{
+					T: 1568404331324,
+					F: 0.013333333333333334,
+					Metric: []labels.Label{
+						{
+							Name:  "filename",
+							Value: `/var/hostlog/apport.log`,
+						},
+						{
+							Name:  "job",
+							Value: "varlogs",
+						},
+					},
+				},
+				{
+					T: 1568404331324,
+					F: 3.45,
+					Metric: []labels.Label{
+						{
+							Name:  "filename",
+							Value: `/var/hostlog/syslog`,
+						},
+						{
+							Name:  "job",
+							Value: "varlogs",
+						},
+					},
+				},
+			},
+			`{
+			"data": {
+			  "resultType": "vector",
+			  "result": [
+				{
+				  "metric": {
+					"filename": "\/var\/hostlog\/apport.log",
+					"job": "varlogs"
+				  },
+				  "value": [
+					1568404331.324,
+					"0.013333333333333334"
+				  ]
+				},
+				{
+				  "metric": {
+					"filename": "\/var\/hostlog\/syslog",
+					"job": "varlogs"
+				  },
+				  "value": [
+					1568404331.324,
+					"3.45"
+				  ]
+				}
+			  ],
+			  "stats" : {
```

Comment:
Nit: I wonder if it's not worth moving this out and replacing with a variable for readability